hazelcast/src/main/java/com/hazelcast/cache/impl/record/CacheRecordHashMap.java

```diff
@@ -36,24 +34,36 @@
     private static final long serialVersionUID = 1L;
 
     private final transient CacheContext cacheContext;
+    private boolean entryCountingEnable;
 
     public CacheRecordHashMap(int initialCapacity, CacheContext cacheContext) {
         super(initialCapacity);
         this.cacheContext = cacheContext;
     }
 
-    public CacheRecordHashMap(int initialCapacity, float loadFactor, int concurrencyLevel,
-            ConcurrentReferenceHashMap.ReferenceType keyType,
-            ConcurrentReferenceHashMap.ReferenceType valueType,
-            EnumSet<Option> options, CacheContext cacheContext) {
-        super(initialCapacity, loadFactor, concurrencyLevel, keyType, valueType, options);
-        this.cacheContext = cacheContext;
+    // Called by only same partition thread. So there is no synchronization and visibility problem.
+    @Override
+    public void setEntryCounting(boolean enable) {
+        if (enable) {
+            if (!entryCountingEnable) {
+                // It was disable before but now it will be enable.
+                // Therefore, we increase the entry count as size of records.
+                cacheContext.increaseEntryCount(size());
+            }
+        } else {
+            if (entryCountingEnable) {
+                // It was enable before but now it will be disable.
+                // Therefore, we decrease the entry count as size of records.
+                cacheContext.decreaseEntryCount(size());
```

Comment:
is there any possibility that we can see `entryCount` negative after this decrease?